mod/storage/pkg/block/store.go

```diff
@@ -86,19 +86,17 @@ func (kv *KVStore[BeaconBlockT]) GetSlotByBlockRoot(
 	return slot, nil
 }
 
-// GetSlotByExecutionNumber retrieves the slot by a given execution number from
+// GetParentSlotByTimestamp retrieves the parent slot by a given timestamp from
 // the store.
-func (kv *KVStore[BeaconBlockT]) GetSlotByExecutionNumber(
-	executionNumber math.U64,
+func (kv *KVStore[BeaconBlockT]) GetParentSlotByTimestamp(
+	timestamp math.U64,
 ) (math.Slot, error) {
-	slot, ok := kv.executionNumbers.Peek(executionNumber)
-	if !ok {
-		return 0, fmt.Errorf(
-			"slot not found at execution number: %d",
-			executionNumber,
-		)
+	slot, _ := kv.timestamps.Peek(timestamp)
+	if slot == 0 {
+		return slot, fmt.Errorf("slot not found at timestamp: %d", timestamp)
 	}
-	return slot, nil
+
+	return slot - 1, nil
```

Comment:
q: seems we are assuming a block is produced every second here. I don't think we can't simply subtract 1 to get parent's block timestamp

Reply:
returns the parent slot for the given timestamp key. as long as slot != 0, which is checked, we should be good to return slot -  1.

Reply:
@calbera , you're right, this is fine!
I have only one concern left: we are assuming that different blocks have different timestamps here.
This is guaranteed by CometBFT for finalized block (see [specs](https://docs.cometbft.com/main/spec/consensus/time)), so I need a pass just to make sure we never store here, non finalized blocks (forks may have the same timestamp and mess things up).

Reply:
store.Set seems to be called upon onFinalizedBlock only so all good

Reply:
sounds good, will include this comment.